### ISSUE TITLE
Create NeuroML2 export from neuromac

### ISSUE THREAD

**Issue by pgleeson** [btorb/neuromac#2]
This is a task which would be useful for the student of @ChristophMetzner, 
Usama Liaquat.

Steps:
- [ ] Install neuromac locally, have a look at the documentation, and get the examples running, e.g. [run_attraction_demo.sh](https://github.com/btorboist/neuromac/blob/master/examples/run_attraction_demo.sh)
- [ ] Have a look and understand the structure of the generated SWC files (see What is SWC? here: http://neuromorpho.org/myfaq.jsp)
- [ ] Fork the neuromac project to your own repo
- [ ] Create a generate_NeuroML2.py based on https://github.com/btorboist/neuromac/blob/master/scripts/generate_SWC_data.py
- [ ] Install jNeuroML, libNeuroML and pyNeuroML
- [ ] Update generate_NeuroML2.py to use libNeuroML to generate the morphologies, see example here: https://github.com/NeuralEnsemble/libNeuroML/blob/master/neuroml/examples/build_3D_network.py
- [ ] Ensure each generated cell has its own `<cell>` definition, and there is a top level `<network>` with one `<population>` for each cell, containing one `<instance>` of that cell located at (0,0,0)
- [ ] Ensure each `<segment>` is part of one of the `<segmentGroup>`s: soma_group, axon_group, dendrite_group; and if it is a dendrite, that it is alos part of one of basal_dendrites, apical_dendrites. See  http://neuromorpho.org/myfaq.jsp
- [ ] Update the example scripts (e.g. run_attraction_demo.sh) to call generate_NeuroML2.py and save the networks to files in each example folder, e.g. demo_attraction/demo_attraction.net.nml.
- [ ] Use:
  
     jnml -validate demo_attraction.net.nml 

to validate the generated files as you go along.
- [ ] Use:
  
     jnml demo_attraction.net.nml -svg
     jnml demo_attraction.net.nml -png

and 

```
pynml-povray -scalez 4  demo_attraction.net.nml
```

To generate images of the nml files.

**Comment by usama57** [comment]
neurocmac is installed now locally and example is also working. I also installed jNeuroML, libNeuroML and pyNeuroML. To create generate_NeuroML2.py in scripts do i have to copy "generate_SWC_data.py"
code in it ? @pgleeson

**Comment by pgleeson** [comment]
@usama57 Yes, start from a copy of generate_SWC_data.py and update that to generate the NeuroML

**Comment by usama57** [comment]
Validating: /home/usama/neuromac/examples/demo_attraction/demo_attraction.net.nml
File: /home/usama/neuromac/examples/demo_attraction/demo_attraction.net.nml is not valid against the schema: /Schemas/NeuroML2/NeuroML_v2beta4.xsd!!
Content is not allowed in prolog.
Validated 1 files: 0 passed, 1 failed
 i am trying to dry run it but it is giving this error. 

i copied this in examples demo_attraction/demo_attraction.net.nml.
 nml_file = 'tmp/net3d.nml'
    writers.NeuroMLWriter.write(nml_doc, nml_file)
    print("Written network file to: "+nml_file) in examples demo_attraction.

**Comment by pgleeson** [comment]
@usama57 Can you show the code you're getting these errors with? Add the generate_NeuroML2.py to your fork and use https://gist.github.com/ to show the generated nml file.

**Comment by usama57** [comment]
@pgleeson what does "name" column in demo_attraction means like "cell_type_1__0", what does it represent? And the other thing, can I use *.SWC file to write a morphology or do I have to write some functions to generate morphology using data in database.
P.s *.swc file contains data generated using database, right?

**Comment by pgleeson** [comment]
@usama57 You'll have to read the paper and have a look at the source code here to figure out what the entries in these files mean. Also, try to be more specific with questions, point to specific lines in committed files. 

Regarding use of SWC, there should be no SWC files read/written in the final generate_NeuroML2.py (but there will be from statements like 'select \* from swc_data'b to access the DB), just use generate_SWC_data.py as a starting point which shows how to access the database to get the points and alter this to generate a neuroml file from the database.

**Comment by usama57** [comment]
@pgleeson  initially i generated with soma and share the result through gist. Kindly check it

**Comment by pgleeson** [comment]
@usama57 Looks like a good start. Can you export the whole cell?

**Comment by usama57** [comment]
@pgleeson  did you mean to export in def run function or through syn db. i read different papers. how to export cell but didn't get any method how to do that. Kindly brief me. DB of demo_attraction only have cell types and i have to export neuroml cell !

**Comment by pgleeson** [comment]
I don't know what you mean by def run function. 

generate_SWC_data.py produces 2 swc files, one for each cell present
generate_NeuroML2.py should produce a single NML2 file containing 2 `<cell>`s, used inside populations in a network. The cells need to have all the correct end points for segments taken from the database
